libs/community/tests/integration_tests/vectorstores/test_aerospike.py

```diff
@@ -0,0 +1,838 @@
+
+        mmr_output = aerospike.max_marginal_relevance_search_by_vector(
+            embedder.embed_query("foo"),
+            index_name=index_name,
+            k=2,
+            fetch_k=3,
+            lambda_mult=0.1,  # more diversity
+        )
+
+        assert len(mmr_output) == 2
+        assert mmr_output[0].page_content == "foo"
+        assert mmr_output[1].page_content == "baz"
+
+        # if fetch_k < k, then the output will be less than k
+        mmr_output = aerospike.max_marginal_relevance_search_by_vector(
+            embedder.embed_query("foo"), index_name=index_name, k=3, fetch_k=2
+        )
+        assert len(mmr_output) == 2
+
+    def test_max_marginal_relevance_search(
+        self, aerospike: Aerospike, admin_client: Any
+    ) -> None:
+        """Test max marginal relevance search."""
+
+        index_name = set_name = get_func_name()
+        admin_client.index_create(
+            namespace=TEST_NAMESPACE,
+            sets=set_name,
+            name=index_name,
+            vector_field=VECTOR_KEY,
+            dimensions=10,
+        )
+        aerospike.add_texts(
+            ["foo", "bar", "baz", "bay", "bax", "baw", "bav"],
+            ids=["1", "2", "3", "4", "5", "6", "7"],
+            index_name=index_name,
+            set_name=set_name,
+        )
+
+        mmr_output = aerospike.max_marginal_relevance_search(
+            "foo", index_name=index_name, k=3, fetch_k=3
+        )
+        sim_output = aerospike.similarity_search("foo", index_name=index_name, k=3)
+
+        assert len(mmr_output) == 3
+        assert mmr_output == sim_output
+
+        mmr_output = aerospike.max_marginal_relevance_search(
+            "foo", index_name=index_name, k=2, fetch_k=3
+        )
+
+        assert len(mmr_output) == 2
+        assert mmr_output[0].page_content == "foo"
+        assert mmr_output[1].page_content == "bar"
+
+        mmr_output = aerospike.max_marginal_relevance_search(
+            "foo",
+            index_name=index_name,
+            k=2,
+            fetch_k=3,
+            lambda_mult=0.1,  # more diversity
+        )
+
+        assert len(mmr_output) == 2
+        assert mmr_output[0].page_content == "foo"
+        assert mmr_output[1].page_content == "baz"
+
+        # if fetch_k < k, then the output will be less than k
+        mmr_output = aerospike.max_marginal_relevance_search(
+            "foo", index_name=index_name, k=3, fetch_k=2
+        )
+        assert len(mmr_output) == 2
+
+    def test_cosine_distance(self, aerospike: Aerospike, admin_client: Any) -> None:
+        """Test cosine distance."""
+        from aerospike_vector_search import types
+
+        index_name = set_name = get_func_name()
+        admin_client.index_create(
+            namespace=TEST_NAMESPACE,
+            sets=set_name,
+            name=index_name,
+            vector_field=VECTOR_KEY,
+            dimensions=10,
+            vector_distance_metric=types.VectorDistanceMetric.COSINE,
```

Comment:
**Security Issue**: The vector_distance_metric is set to COSINE for cosine distance calculation, but there's no validation to ensure that the vectors are normalized which is a prerequisite for cosine similarity calculations. This might lead to incorrect distance calculations if the vectors are not normalized. <br> **Fix**: Ensure that vectors are normalized before performing cosine distance calculations. <br> **Code Suggestion**: 
 ```
 +        # Ensure vectors are normalized
 +        vectors = [normalize_vector(v) for v in vectors]
 +        aerospike.add_texts(
 +            vectors,
 +            ids=["1", "2", "3"],
 +            index_name=index_name,
 +            set_name=set_name,
 +        )  # blocking
 ```